scipy/stats/_continuous_distns.py

```diff
@@ -6121,19 +6122,82 @@ def _entropy(self, c):
     def fit(self, data, *args, **kwds):
         parameters = _check_fit_input_parameters(self, data, args, kwds)
         data, fshape, floc, fscale = parameters
-        if floc is None:
-            return super().fit(data, **kwds)
+
+        ndata = data.shape[0]
+
+        def get_shape(scale, location):
+            # The partial derivative w.r.t. the shape can be solved in
+            # closed form for the shape.
+            return ndata / np.sum(np.log((data - location) / scale))
+
+        if floc is fscale is None:
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed. The equations
+            # are from the partial derivatives of the log-likelihood function.
+
+            def dLdScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return ndata * shape / scale
+
+            def dLdLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize over the scale, setting the partial derivatives
+                # w.r.t. to location and scale equal.
```

Comment:
If we were using [`minimize_scalar`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize_scalar.html), "optimize" would be the appropriate term. 
But we are using `root_scalar`, so we are solving an equation. 
The solution to this equation does minimize/optimize the NLLF, but in the code, we are numerically solving the equation, not performing numerical optimization. 

```suggestion
                # set the partial derivatives w.r.t. to the location and scale
                # equal and solve for the scale
```

Or I guess it would be ok to just change:
```suggestion
                # optimize the scale by setting the partial derivatives
                # w.r.t. to location and scale equal and solving.
```